lib/parachain/collation_protocol_test.go

```diff
@@ -120,3 +122,16 @@ func TestCollationProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestDecodeCollationHandshake(t *testing.T) {
```

Comment:
This isnt actually testing anything is it?

Reply:
It tests that collatorHandshake is an empty struct. (Sure, we know that because we wrote the code. But tests are also used to enforce such behaviour)

We do that by testing that 
- it always encodes to empty bytes (added an extra check for that just now)
- we can decode empty bytes into collatorHandshake.

Reply:
yup this makes sense, I understand the expected behavior of what is being tested better now